dojo_toolkit/test_runner.py

```diff
@@ -0,0 +1,14 @@
+from subprocess import Popen
+
+
+class BaseTestRunner():
+    def __init__(self, code_path):
+        self.code_path = code_path
+
+
+class DoctestTestRunner(BaseTestRunner):
+    def run(self):
+        cmd = "python -m doctest {}/*.py".format(self.code_path)
```

Comment:
I'm trying to apply the [single responsability principle](https://en.wikipedia.org/wiki/Single_responsibility_principle) to the project. Given that the TestRunner should be responsible to only run the tests.

We should have a better mechanism to control message display. The problem in your comment is being currently done by `clock_timer` (what an awful place for that to be). I think we can solve this issue display issue in another PR